core/src/main/scala/cats/Eval.scala

```diff
@@ -110,9 +112,26 @@ case class Now[A](value: A) extends Eval[A]
  * When caching is not required or desired (e.g. if the value produced
  * may be large) prefer Always. When there is no computation
  * necessary, prefer Now.
+ *
+ * Once Later has been evaluated, the closure (and any values captured
+ * by the closure) will not be retained, and will be available for
+ * garbage collection.
  */
 class Later[A](f: () => A) extends Eval[A] {
-  lazy val value: A = f()
+  private[this] var thunk: Function0[A] = f
```

Comment:
Minor style suggestion: could you change the type of `thunk` to `() => A`? Using `() => A` and `Function0[A]` in consecutive lines seems unnecessary confusing.